tests/src/FunctionalJavascript/CivicrmEntityViewsTestBase.php

```diff
@@ -0,0 +1,207 @@
+        'password' => $connection['password'],
+        'host' => $connection['host'],
+        'database' => $connection['database'],
+        'namespace' => $connection['namespace'],
+        'port' => $connection['port'],
+        // CiviCRM does not use prefixes.
+        'prefix' => '',
+      ],
+      'required' => TRUE,
+    ];
+    $this->writeSettings($settings);
+    */
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $admin_user = $this->createUser([
+      'access content',
+      'administer civicrm entity',
+      'administer views',
+    ] + static::$civicrmEntityPermissions);
+    $this->drupalLogin($admin_user);
+    $this->enableCivicrmEntityTypes([static::$civicrmEntityTypeId]);
+
+    // Disable automatic live preview to make the sequence of calls clearer. And
+    // prevent errors on saving the view with the preview ajax load that are
+    // cancelled.
+    \Drupal::configFactory()
+      ->getEditable('views.settings')
+      ->set('ui.always_live_preview', FALSE)
+      ->save();
+  }
+
+  /**
+   * Tests the ViewAddForm for this entity.
+   */
+  public function testAddWizardValues() {
+    $supported_entities = SupportedEntities::getInfo();
+    $civicrm_entity_info = $supported_entities[static::$civicrmEntityTypeId];
+    $this->drupalGet(Url::fromRoute('views_ui.add'));
+
+    $page = $this->getSession()->getPage();
+    $page->fillField('label', static::$civicrmEntityTypeId . ' view');
+    $this->assertJsCondition('jQuery("#edit-label-machine-name-suffix .machine-name-value").html() !== ""');
+
+    $page->selectFieldOption('show[wizard_key]', 'standard:' . static::$civicrmEntityTypeId);
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    if (!empty($civicrm_entity_info['bundle property'])) {
+      $this->assertSession()->optionExists('show[type]', static::$civicrmEntityTypeId);
+    }
+    else {
+      $this->assertSession()->fieldNotExists('show[type]');
+    }
+  }
+
+  /**
+   * Tests creating a basic view with the entity type.
+   */
+  public function testCreateView() {
+    $this->createSampleData();
+
+    $this->drupalGet(Url::fromRoute('views_ui.add'));
+    $page = $this->getSession()->getPage();
+    $page->fillField('label', static::$civicrmEntityTypeId . ' view');
+    $this->assertJsCondition('jQuery("#edit-label-machine-name-suffix .machine-name-value").html() !== ""');
+    $page->selectFieldOption('show[wizard_key]', 'standard:' . static::$civicrmEntityTypeId);
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $page->checkField('page[create]');
+    $page->fillField('page[path]', '/' . static::$civicrmEntityTypeId);
+    $page->pressButton('Save and edit');
+    $this->assertSession()->pageTextContains('The view ' . static::$civicrmEntityTypeId . ' view has been saved.');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->doSetupCreateView();
+    $this->htmlOutput();
+    $page->pressButton('Save');
```

Comment:
😱 of course the assertions failed, I never saved the View after it was configured. Tests green(er)